src/test/java/com/commercetools/project/sync/category/CategorySyncerTest.java

```diff
@@ -19,156 +50,215 @@ void setup() {
-  //        .thenReturn(
-  //            singleton(new ResourceKeyId("categoryKey2",
-  // "ba81a6da-cf83-435b-a89e-2afab579846f")));
-  //    when(targetClient.execute(any(ResourceKeyIdGraphQlRequest.class)))
-  //        .thenReturn(CompletableFuture.completedFuture(resourceKeyIdGraphQlResult));
-  //
-  //    // test
-  //    final CategorySyncer categorySyncer =
-  //        CategorySyncer.of(sourceClient, targetClient, mock(Clock.class));
-  //    categorySyncer.sync(null, true).toCompletableFuture().join();
-  //
-  //    // assertion
-  //    final LoggingEvent errorLog = syncerTestLogger.getAllLoggingEvents().get(1);
-  //    assertThat(errorLog.getMessage())
-  //        .isEqualTo("Warning when trying to sync category. Existing key: categoryKey2");
-  //    assertThat(errorLog.getThrowable().get().getMessage())
-  //        .isEqualTo(
-  //            format(
-  //                "Cannot unset 'orderHint' field of category with id '%s'.",
-  //                sourceCategories.get(0).getId()));
-  //  }
+  @Test
+  void of_ShouldCreateCategorySyncerInstance() {
+    // test
+    final ProjectApiRoot projectApiRoot = mock(ProjectApiRoot.class);
+    final ByProjectKeyCategoriesRequestBuilder byProjectKeyCategoriesRequestBuilder = mock();
+    when(projectApiRoot.categories()).thenReturn(byProjectKeyCategoriesRequestBuilder);
+    final ByProjectKeyCategoriesGet byProjectKeyCategoriesGet = mock();
+    when(byProjectKeyCategoriesRequestBuilder.get()).thenReturn(byProjectKeyCategoriesGet);
+    final CategorySyncer categorySyncer =
+        CategorySyncer.of(projectApiRoot, projectApiRoot, getMockedClock());
+
+    // assertions
+    assertThat(categorySyncer).isNotNull();
+    assertThat(categorySyncer.getQuery()).isEqualTo(byProjectKeyCategoriesGet);
+    assertThat(categorySyncer.getSync()).isExactlyInstanceOf(CategorySync.class);
+  }
+
+  @Test
+  void transform_ShouldReplaceCategoryReferenceIdsWithKeys() throws JsonProcessingException {
+    // preparation
+    final ProjectApiRoot sourceClient = mock(ProjectApiRoot.class);
+    final CategorySyncer categorySyncer =
+        CategorySyncer.of(sourceClient, mock(ProjectApiRoot.class), getMockedClock());
+    final List<Category> categoryPage =
+        asList(
+            readObjectFromResource("category-key-1.json", Category.class),
+            readObjectFromResource("category-key-2.json", Category.class));
+    final List<String> referenceIds =
+        categoryPage.stream()
+            .filter(category -> category.getCustom() != null)
+            .map(category -> category.getCustom().getType().getId())
+            .collect(Collectors.toList());
+
+    final String jsonStringCustomTypes =
+        "{\"data\":{\"typeDefinitions\":{\"results\":[{\"id\":\"53c4a8b4-754f-4b95-b6f2-3e1e70e3d0c3\",\"key\":\"cat1\"}]}}}";
+
+    final GraphQLResponse customTypesResult =
+        readObject(jsonStringCustomTypes, GraphQLResponse.class);
```

Comment:
You could use `JsonUtils.fromJsonString(jsonString, objectType); `

Reply:
For this part I updated the code in the next PR: https://github.com/commercetools/commercetools-project-sync/pull/499/files#diff-dae00fe4ab31d94681d08aae99801537689b5d162abdaaa50d10a46a25a8d429